src/MatrixFields/band_matrix_row.jl

```diff
@@ -0,0 +1,138 @@
+"""
+    BandMatrixRow{ld}(entries...)
+
+Stores the nonzero entries in a row of a band matrix, starting with the lowest
+diagonal, which has index `ld`. Supported operations include accessing the entry
+on the diagonal with index `d` by calling `row[d]`, taking linear combinations
+with other band matrix rows (and with `LinearAlgebra.I`), and checking for
+equality with other band matrix rows (and with `LinearAlgebra.I`). There are
+several aliases defined for commonly-used subtypes of `BandMatrixRow` (with `T`
+denoting the type of the row's entries):
+- `DiagonalMatrixRow{T}`
+- `BidiagonalMatrixRow{T}`
+- `TridiagonalMatrixRow{T}`
+- `QuaddiagonalMatrixRow{T}`
+- `PentadiagonalMatrixRow{T}`
+"""
+struct BandMatrixRow{ld, bw, T} # bw is the bandwidth (the number of diagonals)
+    entries::NTuple{bw, T}
+    BandMatrixRow{ld, bw, T}(entries::NTuple{bw, Any}) where {ld, bw, T} =
```

Comment:
I was hoping to match Julia's default constructor as closely as possible, so that there would be minimal behavioral changes when we get rid of this inner constructor. Specifically, the default constructor is `BandMatrixRow{ld, bw, T}(entries) where {ld, bw, T} = new{ld, bw, T}(convert(NTuple{bw, T}, entries))`. In theory, I could have just replaced `convert` with `rconvert` and left it as is, but I decided to also check that `entries` is already an `NTuple{bw}` just to be safe. If I were to replace the `Any` with `T`, then this would be much more restrictive than the default constructor, and we wouldn't be able to do something like `@. TridiagonalMatrixRow(0, diagonal_field, 0)`, since we'd need to wrap the `0`'s with `FT`s.